### PR TITLE
Feat: Server-side rendering for site frameworks

### PR DESCRIPTION
**Pull request by Meldiron**
## What does this PR do?

Adds SSR support alongside SSG

## Test Plan

Manual; ideally let's add tests

## Related PRs and Issues

x

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [x] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -77,10 +77,10 @@ _APP_COMPUTE_INACTIVE_THRESHOLD=600
 _APP_COMPUTE_MAINTENANCE_INTERVAL=600
 _APP_COMPUTE_RUNTIMES_NETWORK=runtimes
 _APP_EXECUTOR_SECRET=your-secret-key
-_APP_EXECUTOR_HOST=http://proxy/v1
+_APP_EXECUTOR_HOST=http://exc1/v1
```
[the remark this answers is not in the document]
Comment (Meldiron) [reply .env:80]: Keep as is, we need new proxy version that doesnt need `pass` keyword in health endpoint. Aware of it, related to some Cloud changes. This is good enough for now

File: app/config/specs/swagger2-latest-server.json [hunk 2]
```diff
@@ -18034,12 +17970,11 @@
                                     "x-example": "sveltekit",
                                     "enum": [
                                         "sveltekit",
-                                        "nextjs",
-                                        "nuxt",
                                         "astro",
                                         "remix",
-                                        "flutter",
-                                        "static"
+                                        "nuxt",
```
Comment (vermakhushboo) [inline app/config/specs/swagger2-latest-server.json:17975]: Add `other` framework for just HTML files

File: app/config/specs/swagger2-latest-server.json [hunk 3]
```diff
@@ -17575,12 +17575,11 @@
...
```
Comment (vermakhushboo) [inline app/config/specs/swagger2-latest-server.json:17580]: Add `other` for HTML

File: app/config/specs/swagger2-latest-server.json [hunk 4]
```diff
@@ -28548,72 +28421,86 @@
...
+                    "description": "Framework key.",
                     "x-example": "sveltekit"
                 },
                 "name": {
                     "type": "string",
                     "description": "Framework Name.",
                     "x-example": "SvelteKit"
                 },
-                "logo": {
-                    "type": "string",
```
Comment (vermakhushboo) [inline app/config/specs/swagger2-latest-server.json:28560]: Any particular reason for removing logo?

File: src/Appwrite/Platform/Modules/Sites/Http/Sites/CreateSite.php [hunk 5]
```diff
@@ -66,7 +66,7 @@ public function __construct()
             ->param('outputDirectory', '', new Text(8192, 0), 'Output Directory for site.', true)
             ->param('subdomain', '', new CustomId(), 'Unique custom sub-domain. Valid chars are a-z, A-Z, 0-9, period, hyphen, and underscore. Can\'t start with a special char. Max length is 36 chars.', true)
             ->param('buildRuntime', '', new WhiteList(array_keys(Config::getParam('runtimes')), true), 'Runtime to use during build step.')
-            ->param('serveRuntime', '', new WhiteList(array_keys(Config::getParam('runtimes')), true), 'Runtime to use when serving site.')
+            ->param('adapter', '', new Text(8192, 0), 'Framework adapter. Usuallly allows: static, ssr')
```
Comment (vermakhushboo) [inline src/Appwrite/Platform/Modules/Sites/Http/Sites/CreateSite.php:69]: Should it be a whitelist validator instead?

File: app/config/specs/swagger2-latest-server.json (hunk 4 above)
Comment (Meldiron) [reply app/config/specs/swagger2-latest-server.json:28560]: Was never actually sent from server

File: src/Appwrite/Platform/Modules/Sites/Http/Sites/CreateSite.php (hunk 5 above)
Comment (Meldiron) [reply src/Appwrite/Platform/Modules/Sites/Http/Sites/CreateSite.php:69]: We cant have whitelist yet, the whitelist values are only defined after we know framework. I added validator first thing inside the action

Comment (Meldiron) [reply src/Appwrite/Platform/Modules/Sites/Http/Sites/CreateSite.php]: I opened disucssion with team leads, Ill update it if possible in-line. But probably not, so this is good enough.

**Review by vermakhushboo — APPROVED** [review]

**Pull request merged**